src/custom/hooks/useChangeNetworks.ts

```diff
@@ -0,0 +1,75 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import { UnsupportedChainIdError, useWeb3React } from '@web3-react/core'
+import { useOnClickOutside } from 'hooks/useOnClickOutside'
+import { useActiveWeb3React } from 'hooks/web3'
+import { useModalOpen, useToggleModal, useWalletModalToggle } from 'state/application/hooks'
+import { ApplicationModal } from 'state/application/reducer'
+import { useAppSelector } from 'state/hooks'
+import { CHAIN_INFO, SupportedChainId } from 'constants/chains'
+import { switchToNetwork } from 'utils/switchToNetwork'
+import { supportedChainId } from 'utils/supportedChainId'
+
+export default function useChangeNetworks() {
+  const { chainId: preChainId, library, account } = useActiveWeb3React()
+  const { error } = useWeb3React() // MOD: check unsupported network
+  const node = useRef<HTMLDivElement>()
+  const open = useModalOpen(ApplicationModal.NETWORK_SELECTOR)
+  const toggle = useToggleModal(ApplicationModal.NETWORK_SELECTOR)
```

Comment:
maybe give it a better name? i was in line 36 trying to guess what it toggles

Reply:
this is all old code i just pasted